worker/changestream/eventmultiplexer/eventmultiplexer.go

```diff
@@ -239,6 +257,8 @@ func (e *EventMultiplexer) loop() error {
 			if err := e.dispatchSet(changeSet); err != nil {
 				e.logger.Errorf("dispatching set: %v", err)
 				e.dispatchErrorCount++
+				// Increment number of dispatch errors metric.
+				e.metrics.DispatchErrorsInc()
```

Comment:
I think we could try and pass the error here and in the metrics collector factor out common ones.

Reply:
Checking `Subscription.dispatch()` I only see `tomb.ErrDying` possible, is it useful? or am I missing some other errors

Reply:
If you were to look at this from the mertics page, I guarantee the next question you'll ask yourself when you see this, is why?

The alternative solution is to say in the description about the failure modes. The only thing I have with that solution is that it's less discoverable.